src/globus_sdk/services/auth/client/service_client.py

```diff
@@ -631,3 +681,1111 @@ def delete_project(self, project_id: UUIDLike) -> GlobusHTTPResponse:
+        required_idp: UUIDLike | None | utils.MissingType = utils.MISSING,
+        preselect_idp: UUIDLike | None | utils.MissingType = utils.MISSING,
+        additional_fields: dict[str, t.Any] | utils.MissingType = utils.MISSING,
+    ) -> GlobusHTTPResponse:
+        """
+        Update a client. Requires the ``manage_projects`` scope.
+
+        :param client_id: ID of the client to update
+        :type client_id: str or uuid
+        :param name: The display name shown to users on consents. May not contain
+            linebreaks.
+        :type name: str
+        :param visibility: If set to "public", any authenticated entity can view it.
+            When set to "private", only entities in the same project as the client can
+            view it.
+        :type visibility: str
+        :param redirect_uris: list of URIs that may be used in OAuth authorization
+            flows.
+        :type redirect_uris: iterable of str
+        :param terms_and_conditions: URL of client's terms and conditions.
+        :type terms_and_conditions: str
+        :param privacy_policy: URL of client's privacy policy.
+        :type privacy_policy: str
+        :param required_idp: In order to use this client a user must have an identity
+            from this IdP in their identity set.
+        :type required_idp: str or uuid
+        :param preselect_idp: This pre-selects the given IdP on the Globus Auth login
+            page if the user is not already authenticated.
+        :type preselect_idp: str or uuid
+        :param additional_fields: Any additional parameters to be passed through.
+        :type additional_fields: dict
+
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                When creating a project, your account is not necessarily included as an
+                admin. The following snippet uses the ``manage_projects`` scope as well
+                as the ``openid`` and ``email`` scopes to get the current user ID and
+                email address and use those data to setup the project.
+
+                .. code-block:: pycon
+
+                    >>> ac = globus_sdk.AuthClient(...)
+                    >>> client_id = ...
+                    >>> r = ac.create_update(client_id, name="Foo Utility")
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: auth.update_client
+
+            .. tab-item:: API Info
+
+                ``POST /v2/api/clients/{client_id}``
+
+                .. extdoclink:: Update Client
+                    :ref: auth/reference/#update_client
+        """
+        body: dict[str, t.Any] = {
+            "name": name,
+            "visibility": visibility,
+            "redirect_uris": redirect_uris,
+        }
+
+        if required_idp is not utils.MISSING:
+            body["required_idp"] = str(required_idp)
+        if preselect_idp is not utils.MISSING:
+            body["preselect_idp"] = str(preselect_idp)
```

Comment:
I'm losing count but I think we're now at ~6 places where auto-stringifying UUIDs would make this smoother. And I'm sure it's only going to go up. So I'm definitely going to give that a look after this PR.